pdns/recursordist/lwres.cc

```diff
@@ -382,11 +483,106 @@ static void addPadding(const DNSPacketWriter& pw, size_t bufsize, DNSPacketWrite
   }
 }
 
+static void outgoingCookie(const OptLog& log, const ComboAddress& address, const timeval& now, DNSPacketWriter::optvect_t& opts, std::optional<EDNSCookiesOpt>& cookieSentOut, std::optional<ComboAddress>& addressToBindTo)
+{
+  auto lock = s_cookiestore.lock();
+  auto found = lock->find(address);
+  if (found != lock->end()) {
+    switch (found->getSupport()) {
+    case CookieEntry::Support::Supported:
+    case CookieEntry::Support::Probing:
+      cookieSentOut = found->d_cookie;
+      addressToBindTo = found->d_localaddress;
+      opts.emplace_back(EDNSOptionCode::COOKIE, cookieSentOut->makeOptString());
+      found->d_lastupdate = now.tv_sec;
+      VLOG(log, "Sending stored cookie info to " << address.toString() << ": " << found->d_cookie.toDisplayString() << endl);
+      break;
+    case CookieEntry::Support::Unsupported:
+      VLOG(log, "Server " << address.toString() << " does not support cookies" << endl);
+      break;
+    }
+  }
+  else {
```

Comment:
nit: I prefer early `return` instead of the whole length of the function to be `if/else`